test/python/transpiler/test_reset_after_measure_simplification.py

```diff
@@ -0,0 +1,141 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the ResetAfterMeasureSimplification pass"""
+
+from qiskit import QuantumCircuit, ClassicalRegister, QuantumRegister
+from qiskit.circuit.classicalregister import Clbit
+from qiskit.transpiler.passes.optimization import ResetAfterMeasureSimplification
+from qiskit.test import QiskitTestCase
+
+
+class TestResetAfterMeasureSimplificationt(QiskitTestCase):
+    """Test ResetAfterMeasureSimplification transpiler pass."""
+
+    def test_simple(self):
+        """Test simple"""
+        qc = QuantumCircuit(1, 1)
+        qc.measure(0, 0)
+        qc.reset(0)
+
+        new_qc = ResetAfterMeasureSimplification()(qc)
+
+        ans_qc = QuantumCircuit(1, 1)
+        ans_qc.measure(0, 0)
+        ans_qc.x(0).c_if(ans_qc.clbits[0], 1)
+        new_qc.draw("mpl", filename="/tmp/foo.png")
```

Comment:
Hard-coded temp file name is causing the test failures.  Probably something like
```python
with tempfile.NamedTemporaryFile(suffix=".png") as file:
    new_qc.draw("mpl", filename=file.name)
```
I think?

Reply:
oops my bad, this was from me debugging an earlier version of the patch where I forgot to add the cargs on the measurement in the subgraph we replaced and I couldn't figure out what I did wrong. I'll remove this

Reply:
Removed in: https://github.com/Qiskit/qiskit-terra/pull/8330/commits/b3f767a4781283dd2578e4a9386b54aa08f6acd2